vignettes/miaViz.Rmd

```diff
@@ -326,6 +326,100 @@ names(metadata(dmn_se))
 plotDMNFit(dmn_se, type = "laplace")
 ```
 
+# Serial data ordination and trajectories
+
+```{r, error=FALSE, warning=FALSE, results='hide'}
+if (!requireNamespace("devtools", quietly = TRUE))
+    install.packages("devtools")
+devtools::install_github("microbiome/miaTime")
+```
+
+Principal Coordinates Analysis using Bray-Curtis dissimilarity on the
+`hitchip1006` dataset:
+
+```{r}
+library(dplyr)
+data(hitchip1006, package = "miaTime")
+tse <- hitchip1006
+tse <- transformCounts(tse, method = "relabundance")
+## Ordination with PCoA with Bray-Curtis dissimilarity
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+                  name = "PCoA_BC", exprs_values = "relabundance",
+              na.rm = TRUE)
+# Calculating Explained variance
+e <- attr(reducedDim(tse, "PCoA_BC"), "eig")
+rel_eig <- e/sum(e[e>0])
+# plot
+p <- plotReducedDim(tse, dimred = "PCoA_BC")
+
+p + labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]],1), "%", ")", sep = ""),
+         y = paste("PCoA 2 (", round(100 * rel_eig[[2]],1), "%", ")", sep = ""),
+         title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+```
+
+Retrieving information about all available trajectories:
+
+```{r}
+# List subjects with two time points
+selected.subjects <- names(which(table(tse$subject)==2))
+
+# Subjects counts per number of time points available in the data
+table(table(tse$subject)) %>% as.data.frame() %>%
+    rename(Timepoints=Var1, Subjects=Freq)
+```
+
+Lets look at all trajectories having two time points in the data:
+
+```{r}
+# adding metadata for the plot
+p$data <- cbind(p$data, colData(tse))
+# plot
+p + geom_path(aes(x=X, y=Y, group=subject), 
+              arrow=arrow(length = unit(0.1, "inches")),
+              data = subset(p$data, subject %in% selected.subjects))+
+    labs(title = "All trajectories with two time points")+
+    theme(plot.title = element_text(hjust = 0.5))
```

Comment:
Just a comment: Do you know what determines the direction of arrow? It seems that you haven't done that explicitly --> is the direction always from time point 1 to time point 2?

Reply:
[arrow(ends="last")]) does that. However "last" refers to the latter observation in the table (not necessarily the latter time point)

However, because time points are in increasing order in your data it seems to be right, but you might want to order the data manually to be sure (and for example purpose)

Reply:
Thanks for clarification. 

The comment about time series sorting is important because this is something a naive user might very easily overlook. In the present example the solution needs to take also other considerations into account. It works for this case but also makes the example more complex.

Let us see if the example can be simplified. 1) The coloring in geom_path is here mentioned as a motivation for using reverse arrow solution but in the code above does not color the segment/arrow (or does this coloring come in some later example?). 2) The rate of change (or divergence) in the segment is determined by the distance between two points, and this forms its own column in colData; the coordinates of the start and points should not be relevant for the geom_path coloring if the divergence has been calculated and added to colData. 

I would support the idea to sort samples by time, and use default arrow (not reversed). Then coloring the arrow based on the additional time_divergence colData field if appropriate. If this could be done, the example would be more easy to understand and less error-prone for naive users. But is there some issue that I have missed and prevents simplifying the example like this?

Reply:
1. At next example comes the coloring.
2. Eventually I calculated the divergences with getstepwisedivergence, and results were stored in colData, and initial time points for each series obtained were NAs  which is expected. I tested the coloring with geom_path and the order of the points mattered, it picked always the first point in the series (or first row of the grouped subject) then colored the corresponding segment with it. For this reason I reverse sorted the points and the arrow end also. 
I double checked all these, but I could have perhaps missed smt on the way. If somebody would like to test or an easier way of doing these, suggestions are very welcome.

Reply:
The getStepwiseDivergence adds two fields to colData: time_difference and time_divergence (those are the default names). The time_difference is NA for subjects with just one time point, or for subjects with missing time information in the calculation. The getStepwiseDivergence function should not affect the actual time point information (first or second time point). 

Can you show a example that clarifies the change in time point information following the application of getStepwiseDivergence?

Reply:
Ah it comes in next comments, no need for action here (except that we need to resolve how to do this).